config/src/main/java/org/springframework/security/config/annotation/web/configuration/WebSecurityConfiguration.java

```diff
@@ -99,16 +98,6 @@ public SecurityExpressionHandler<FilterInvocation> webSecurityExpressionHandler(
 	 */
 	@Bean(name = AbstractSecurityWebApplicationInitializer.DEFAULT_FILTER_NAME)
 	public Filter springSecurityFilterChain() throws Exception {
-		boolean hasConfigurers = this.webSecurityConfigurers != null && !this.webSecurityConfigurers.isEmpty();
-		boolean hasFilterChain = !this.securityFilterChains.isEmpty();
-		Assert.state(!(hasConfigurers && hasFilterChain),
-				"Found WebSecurityConfigurerAdapter as well as SecurityFilterChain. Please select just one.");
-		if (!hasConfigurers && !hasFilterChain) {
-			WebSecurityConfigurerAdapter adapter = this.objectObjectPostProcessor
-					.postProcess(new WebSecurityConfigurerAdapter() {
-					});
-			this.webSecurity.apply(adapter);
```

Comment:
Does this removal affect non-Spring Boot usage of the framework? I'm wondering if an XML-based application would no longer get a default Spring Security configuration?

Reply:
Good catch, I've removed way too much from this method without realizing it. I will fix it and some tests that don't need an explicit `SecurityFilterChain` bean